### ISSUE TITLE
Vue dynamic paths

### ISSUE THREAD

**Issue by LukeeeeBennett** [elastic/apm-agent-rum-js#510]
I have a vue app that uses vue-router. A large section of the app uses dynamic routes, the dynamic parameter in the url is used to register/render a specific dynamic component.

Because of this, I would rather name the transaction after the fully resolved path rather than the path with the parameter syntax. I can see how for `/users/:id` it may be useful, but should I look to resolve `/page/:component_type` to `/page/spaceship`?

I achieved it with a  simple router hook.

```
router.beforeEach((to, from, next) => {
  Vue.prototype.$apm.getCurrentTransaction().name = to.path
  next()
})
```

If this is a correct assumption, would you be open to an upstream contribution to support this within the vue package?

Options:
1. Add route-specific config. .e.g `{ path: '/page/:dynamic', $apm: { shouldResolvePath: true }}` (not sure how you feel extending a route config)
2. Add plugin config array. .e.g `{ config: { resolvePaths: ['/page/:dynamic'] } }`
3. Add plugin config function. e.g. `{ config: { transactionName(to, from) { return to.path } } }` (flexible but more work to implement if you want route-specific)

**Comment by vigneshshanmugam** [comment]
The reason for using the parameter/slug pattern is to avoid too many transactions on the UI (cardinality issue) that would refer to the same page and would be hard for the user to track the reponse times of the same transaction. 

By using the slug, pattern we can group all similar transactions and the UI would also handle it smoothly. We have the exact pathname information of the transaction under the transaction context `context.page.url`

I am happy to extend the config options of the router to allow for the user's needs but would that help you solve your problem?

**Comment by PaulMougel** [comment]
FWIW, we have opted in our Vue app to name transactions after route names (`to.name`) instead of route paths. As we use [named routes](https://router.vuejs.org/guide/essentials/named-routes.html), we found it to be more precise and less prone to errors.

**Comment by vigneshshanmugam** [comment]
@PaulMougel Thats interesting, But will all the routes have name predefined and is it a required parameter? One of the reasons we went with path is to get be close to the URL navigation a possible. But I would be interested to know more about the decision to use `name` and also we can add support to the Vue Plugin if it makes the usage simpler for the users. 

We are also to Pull Requests and happy to help as well.

**Comment by PaulMougel** [comment]
Route names are optional (as described in [the documentation of Vue Router](https://router.vuejs.org/api/#routes). In our app all routes have names by convention, but from what I understand not all apps using Vue Router will. Maybe we could use the route name if it exist and fallback on the URL?

As we use route names in our code (when using programmatic navigation or with the `router-link` component), we thought monitoring them in APM would make debugging easier.

**Comment by vigneshshanmugam** [comment]
Yeah for sure that would totally work and as you mentioned could help debug issues faster. Would be interested in opening a PR? Should be easy to do it here https://github.com/elastic/apm-agent-rum-js/blob/master/packages/rum-vue/src/route-hooks.js#L32-L47
